Makefile

```diff
@@ -80,6 +89,84 @@ logs.%: ## Shows the logs of the given docker service. Example: make logs.datast
 restart.%: ## Restarts the given docker service. Example: make restart.datastore
 	docker compose restart $*
 
-run-backend-tests: ## [Not ready for use] Runs the backend tests
-	@echo "Run the backend test on the following module: $(RUN_ARGS)"
-	@echo "Not in use, under construction!"
+run_tests.backend: ## [Not ready for use] Runs the backend tests
+	docker compose up datastore redis -d
+	docker compose run --no-deps --entrypoint "python -m scripts.run_backend_tests $(PYTHON_ARGS)" dev-server
+	$(MAKE) stop
+
+run_tests.frontend: ## Runs the frontend unit tests
+	docker compose run --no-deps --entrypoint "python -m scripts.run_frontend_tests $(PYTHON_ARGS)" dev-server
+
+run_tests.typescript: ## Runs the typescript checks
+	docker compose run --no-deps --entrypoint "python -m scripts.typescript_checks" dev-server
+
+run_tests.custom_eslint: ## Runs the custome eslint tests
+	docker compose run --no-deps --entrypoint "python -m scripts.run_custom_eslint_tests" dev-server
+
+run_tests.mypy: ## Runs mypy checks
+	docker compose run --no-deps --entrypoint "python -m scripts.run_mypy_checks" dev-server
+
+run_tests.backend_associate: ## Runs the backend associate tests
+	docker compose run --no-deps --entrypoint "python -m scripts.check_backend_associated_test_file" dev-server
+
+run_tests.acceptance: ## Runs the acceptance tests for the parsed suite
+	@echo 'Shutting down any previously started server.'
+	$(MAKE) stop 
+	$(MAKE) start-devserver-for-tests
+	@echo 'Starting acceptance test for the suite: $(suite)'
+	./node_modules/.bin/jasmine --config="./core/tests/puppeteer-acceptance-tests/jasmine.json" ./core/tests/puppeteer-acceptance-tests/spec/$(suite)
+	$(MAKE) stop
+
+CHROME_VERSION := $(shell google-chrome --version | awk '{print $$3}')
+
+run_tests.e2e: ## Runs the e2e tests for the parsed suite
+	@echo 'Shutting down any previously started server.'
+	$(MAKE) stop 
+	$(MAKE) start-devserver-for-tests
+	@echo 'Starting e2e test for the suite: $(suite)'
+	../oppia_tools/node-v16.13.0-linux-x64/bin/npx wdio ./core/tests/wdio.conf.js --suite $(suite) $(CHROME_VERSION) --params.devMode=True --capabilities[0].maxInstances=3
+	$(MAKE) stop
+
+OS_NAME := $(shell uname)
+
+.PHONY: install_node
+
+install_node:
+	@echo "Installing Node.js..."
+	@if [ "$(OS_NAME)" = "Windows" ]; then \
+		if [ "$(shell python -c 'import sys; print(sys.maxsize > 2**32)')" = "True" ]; then \
```

Comment:
What if python is not installed? Can you try doing this via the terminal in windows?

Reply:
um, I dont think it would be easier for me to test on a windows system.

Reply:
Can we do something similar to this? https://techgenix.com/aquicktiptocheckoperatingsystemarchitecture/

Reply:
https://github.com/oppia/oppia/issues/18638
added in the future enhancements

Reply:
added to the future enhancement issue with the top-priority

Reply:
resolved